soh/soh/Enhancements/mods.cpp

```diff
@@ -219,51 +210,43 @@ void RegisterFreezeTime() {
 }
 
 /// Switches Link's age and respawns him at the last entrance he entered.
-void RegisterSwitchAge() {
-    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnGameFrameUpdate>([]() {
-        static bool warped = false;
+void SwitchAge() {
+    if (gPlayState == NULL) return;
 
-        if (!GameInteractor::IsSaveLoaded(true)) {
-            CVarClear(CVAR_GENERAL("SwitchAge"));
-            warped = false;
-            return;
-        }
-
-        static Vec3f playerPos;
-        static int16_t playerYaw;
-        static RoomContext* roomCtx;
-        static s32 roomNum;
-
-        if (CVarGetInteger(CVAR_GENERAL("SwitchAge"), 0) && !warped) {
-            playerPos = GET_PLAYER(gPlayState)->actor.world.pos;
-            playerYaw = GET_PLAYER(gPlayState)->actor.shape.rot.y;
-            roomCtx = &gPlayState->roomCtx;
-            roomNum = roomCtx->curRoom.num;
-            ReloadSceneTogglingLinkAge();
-            warped = true;
-        }
+    Player* player = GET_PLAYER(gPlayState);
 
-        if (warped && gPlayState->transitionTrigger != TRANS_TRIGGER_START &&
-            gSaveContext.nextTransitionType == TRANS_NEXT_TYPE_DEFAULT) {
-            GET_PLAYER(gPlayState)->actor.shape.rot.y = playerYaw;
-            GET_PLAYER(gPlayState)->actor.world.pos = playerPos;
-            if (roomNum != roomCtx->curRoom.num) {
-                func_8009728C(gPlayState, roomCtx, roomNum); //load original room
-                //func_800973FC(gPlayState, &gPlayState->roomCtx); // commit to room load?
-                func_80097534(gPlayState, roomCtx);  // load map for new room (unloading the previous room)
-            }
-            warped = false;
-            CVarClear(CVAR_GENERAL("SwitchAge"));
-        }
+    gPlayState->nextEntranceIndex = gSaveContext.entranceIndex;
+    gPlayState->transitionTrigger = TRANS_TRIGGER_START;
+    gPlayState->transitionType = TRANS_TYPE_INSTANT;
+    if (
+        // Hyrule Castle: Very likely to fall through floor
+        gPlayState->sceneNum != SCENE_HYRULE_CASTLE &&
+        gPlayState->sceneNum != SCENE_OUTSIDE_GANONS_CASTLE &&
```

Comment:
For the Castle edge case, using child Link's pos for adult Link respawn places adult link in the loading zone and causes him to travel back to market.

Similarly, having exiting the castle or fairy fountain as the last entrance ends up with opposite ages spawning in voids/undesirable locations.

Reply:
Maybe for hyrule castle specifically we just have a mapping of valid entranceId conversions?

Reply:
I was thinking we can just hardcode to the front of the scene right now, or force the entrance to reload.